src/core_plugins/kibana/public/visualize/editor/agg_params.js

```diff
@@ -82,11 +82,23 @@ uiModules
         // build collection of agg params html
         type.params.forEach(function (param, i) {
           let aggParam;
+          let fields;
+          // if field param exists, compute allowed fields
+          if (param.name === 'field') {
+            fields = $aggParamEditorsScope.indexedFields;
+            $aggParamEditorsScope.showAnalyzedWarning = true;
+            if (!_.isUndefined(param.showAnalyzedWarning) && !param.showAnalyzedWarning) {
+              // if not explicitly set, always show the warning about analyzed field
+              $aggParamEditorsScope.showAnalyzedWarning = false;
```

Comment:
This is what I tried to say with a way too short sentence, sorry:

- the `agg.params.field` object only gives me information about a field, e.g., name, type, etc. 
- instead, the `showAnalyzedWarning` option is accessible in `$scope.agg.type.params`
- however, that variable is an array, and so I have to iterate it to find the `field` param, through which I can access the parameter. Therefore, I add that parameter to the editor scope.

However, this could have been simplified with what you said by setting the default in `FieldParam`: 
https://github.com/elastic/kibana/pull/7302/commits/0d924fc73d6ebc907ce3ff6a73b2b5950a1ca9ec